### ISSUE TITLE
UX - Ability to tag and filter images in image library

### ISSUE THREAD

**Issue by stpaultim** [backdrop/backdrop-issues#5680]
## Description of the need

I am just finishing a Drupal 7 to Backdrop CMS upgrade that has approximately 3500 managed images in the image library with only one way to sort or organize them, by name. I immediately began to think about how the site admins better manage this number of images and quickly find images that they might want to re-use for specific purposes. 

Examples:

- Photos of staff for use in organizational updates
- Images suited for hero block at top of page
- Images about a specific program

I recently had a similar use case with a Wordpress site and found a number of paid plug-ins that provide folders or tagging for images. 

At first glance, it would appear to many beginners that Backdrop does not support this. I suspect that many of them would go looking for a contrib module to help with this. In fact, a pretty elegant solution is available in Backdrop core, but I suspect difficult to find for many new users. 

1) Add an Image Category vocabulary (or use existing 'tags'). 
2) Add a term reference field to image file type.
3) Modify 'Image Library' view that comes with core to include an exposed filter for tags.

## Proposed solution

I propose adding at least #2 and #3 (using existing "tags" vocabulary) or adding #1, #2, and #3. 

In my view, this relatively simple change to the install profile would address a fairly common use case and make the existing image libary easier to use and more powerful. 

UI for adding in image would look something like this (but with a little more styling):
![image](https://user-images.githubusercontent.com/3144571/178128731-a190ae13-c803-4ab2-83f7-1c709dabdebe.png)

## Alternatives that have been considered

This can be configured by any site architect, but the solution is not easy to see or find and I suspect many site architects may not find this solution without a little help. 

We could also look at a contrib module or recipe that adds this feature. 

## Draft of feature description for Press Release (1 paragraph at most)

Backdrop now includes the ability to tag images by type or category and comes with an additional filter to help manage your files. This will be particulary helpful on sites with lots of images.

**Comment by indigoxela** [comment]
I belief, something like that has been asked/requested before, but don't seem to find an existing issue for that. (Maybe it has just been discussed in our chat?).

Yes, with *lots* of images, the image library quickly gets cumbersome.

> This can be configured by any site architect

IMO that's an important point. Not all sites will have thousands of images - probably only a minority.

Another problem: while File entities are fieldable, these fields won't be available to add via "regular" image field upload. So I'm assuming, it site architects need to consider the whole upload workflow.

> We could also look at a contrib module or recipe that adds this feature

My current suspicion is: it's a contrib candidate. Maybe I could get convinced, though, that it's not. :wink:

**Comment by argiepiano** [comment]
I agree this would be a good core addition. As @indigoxela pointed above, the fact that this tags field will not currently be available through the file field widget is a problem. As a workaround, currently you would be able to tag the images if you uploaded it through the file add UI at file/add

**Comment by muranod** [comment]
I'm trying to implement this now on my own site, using what is currently available. My photography site is about showing slideshows, often with 20 or so images, but then for users to be able to pull up all images that fit a certain category, such as cats, dogs, landscapes, etc. I added a term field to the images (in a sandbox) , but don't see any way to add the tags during upload. That's not a big problem, since I can add the tags after the fact. 

I guess the best way to implement this for site visitors would be to create and attach a custom block to the page with a list of terms that would pull up collections of images. I think this is all doable with what's out there.  That said, it would be helpful to be able to add the term/tags as the images are being added to the content node. Don't see a way to do that currently.
